lib/rs-address.ts

```diff
@@ -0,0 +1,375 @@
+
+        for (var i = 0; i < 17; i++) {
+            out += this.alphabet[this.codeword[this.cwmap[i]]];
+
+            if ((i & 3) == 3 && i < 13) out += '-';
+        }
+
+        return out;
+    } //__________________________
+
+    account_id() {
+        var out = '',
+            inp = [],
+            len = 13;
+
+        for (var i = 0; i < 13; i++) {
+            inp[i] = this.codeword[12 - i];
+        }
+
+        do // base 32 to base 10 conversion
+        {
+            var divide = 0,
+                newlen = 0;
+
+            for (i = 0; i < len; i++) {
+                divide = divide * 32 + inp[i];
+
+                if (divide >= 10) {
+                    inp[newlen++] = Math.floor(divide / 10);
+                    divide %= 10;
+                } else if (newlen > 0) {
+                    inp[newlen++] = 0;
+                }
+            }
+
+            len = newlen;
+            out += String.fromCharCode(divide + '0'.charCodeAt(0));
+        }
+        while (newlen);
+
+        return out.split("").reverse().join("");
+    } //__________________________
+
+    set(adr: string, allow_accounts?: any) {
+        if (typeof allow_accounts === 'undefined') allow_accounts = true;
+
+        var len = 0;
+        this.guess = [];
+        this.reset();
+
+        adr = String(adr);
+
+        adr = adr.replace(/(^\s+)|(\s+$)/g, '').toUpperCase();
+
+        if (adr.indexOf("GMD-") == 0) adr = adr.substr(4);
+
+        if (adr.match(/^\d{1,20}$/g)) // account id
+        {
+            if (allow_accounts) return this.from_acc(adr);
+        } else // address
+        {
+            let clean = [];
+
+            for (var i = 0; i < adr.length; i++) {
+                var pos = this.alphabet.indexOf(adr[i]);
+
+                if (pos >= 0) {
+                    clean[len++] = pos;
+                    if (len > 18) return false;
+                }
+            }
+        }
+        let clean: any[] = [];
+        if (len == 16) // guess deletion
+        {
+            for (let i = 16; i >= 0; i--) {
+                for (var j = 0; j < 32; j++) {
+                    clean[i] = j;
+
+                    this.set_codeword(clean);
+
+                    if (this.ok()) this.add_guess();
+                }
+
+                if (i > 0) {
+                    let t = clean[i - 1];
+                    clean[i - 1] = clean[i];
+                    clean[i] = t;
+                }
+            }
+        }
+
+        if (len == 18) // guess insertion
+        {
+            for (let i = 0; i < 18; i++) {
+                this.set_codeword(clean, 18, i);
+
+                if (this.ok()) this.add_guess();
+            }
+        }
+
+        if (len == 17) {
+            this.set_codeword(clean);
+
+            if (this.ok()) return true;
+
+            if (this.guess_errors() && this.ok()) this.add_guess();
+        }
+
+        this.reset();
+
+        return false;
+    }
+}
+
+export default RSAddress;
```

Comment:
Maybe some more types in this class if you feel they are needed?

Reply:
Same comment as above: https://github.com/CoopNetwork/GMD-nodejs-sdk/pull/3#discussion_r961358164